controllers/subscription/subscription_controller.go

```diff
@@ -246,40 +258,53 @@ func (r *SubscriptionReconciler) HandleUpgrades(ctx context.Context, rhmiSubscri
 		}
 	}
 
-	isInstallPlanDeleted := false
-	currentOperatorVersionName := fmt.Sprintf("%s.v%s", CSVNamePrefix, version.GetVersion())
+	csvNamePrefix := "integreatly-operator"
+	if installation.Spec.Type == string(integreatlyv1alpha1.InstallationTypeManagedApi) {
+		csvNamePrefix = "managed-api-service"
+	}
+
+	currentOperatorVersionName := fmt.Sprintf("%s.v%s", csvNamePrefix, version.GetVersion())
 	if csvFromCatalogSource.Spec.Replaces != currentOperatorVersionName {
+		err = rhmiConfigs.DeleteInstallPlan(ctx, latestRHMIInstallPlan, r.Client)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("error deleting installplan %w", err)
+		}
 
-		if csvFromCatalogSource.Spec.Replaces != latestRHMICSV.Spec.Replaces {
-			err = rhmiConfigs.DeleteInstallPlan(ctx, latestRHMIInstallPlan, r.Client)
-			if err != nil {
-				return ctrl.Result{}, fmt.Errorf("error deleting installplan %w", err)
-			}
+		log.Info("Installplan deleted for the install of patch upgrade")
 
-			isInstallPlanDeleted = true
-			log.Info("Installplan deleted for the install of patch upgrade")
+		err = rhmiConfigs.CreateInstallPlan(ctx, rhmiSubscription, r.Client)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
 
-			err = rhmiConfigs.CreateInstallPlan(ctx, rhmiSubscription, r.Client)
+		err := wait.Poll(time.Second*5, time.Minute*1, func() (done bool, err error) {
+			// gets the subscription with the recreated installplan
+			logrus.Infof("Entered wait.Poll")
+			err = r.Client.Get(ctx, k8sclient.ObjectKey{Name: rhmiSubscription.Name, Namespace: rhmiSubscription.Namespace}, rhmiSubscription)
+			logrus.Infof("got subscription")
 			if err != nil {
-				return ctrl.Result{}, err
+				log.Infof("Couldn't get subscription", l.Fields{"Error": err})
+				return false, nil
 			}
-		}
-
-		if isInstallPlanDeleted {
-			// Requeue reconciler until the installplan is recreated
-			return ctrl.Result{
-				Requeue:      true,
-				RequeueAfter: 10 * time.Second,
-			}, nil
-		}
-
-		// gets the subscription wit the recreated installplan
-		err := r.Client.Get(ctx, k8sclient.ObjectKey{Name: rhmiSubscription.Name, Namespace: rhmiSubscription.Namespace}, rhmiSubscription)
+			logrus.Infof("InstallPlan read from the new subscription")
+			if rhmiSubscription.Status.InstallPlanRef == nil {
```

Comment:
this verification is done inside the [GetLatestInstallPlan](https://github.com/integr8ly/integreatly-operator/blob/fe9ad0579ace958ad901ba9b5c8ee068eae411de/controllers/subscription/rhmiConfigs/main.go#L31) method, so I believe we can remove this if statement